share/functions/__fish_svn_prompt.fish

```diff
@@ -0,0 +1,169 @@
+set -g __fish_svn_prompt_flag_names added conflicted deleted ignored modified replaced unversioned_external unversioned missing locked scheduled switched token_present token_other token_stolen token_broken
+
+function __fish_svn_prompt_parse_status --argument flag_status_string --description "helper function that does pretty formatting on svn status"
+	# iterate over the different status types
+	for flag_index in (seq (count $__fish_svn_prompt_flag_names))
+		# resolve the name of the variable for the character representing the current status type
+		set -l flag_name __fish_svn_prompt_char_{$__fish_svn_prompt_flag_names[$flag_index]}_char
+		# check to see if the status string for this column contains the character representing the current status type
+		if test (echo $flag_status_string | grep -c $$flag_name) -eq 1
+			# if it does, then get the names of the variables for the display character and colour to format it with
+			set -l flag_display __fish_svn_prompt_char_{$__fish_svn_prompt_flag_names[$flag_index]}_display
+			set -l flag_color __fish_svn_prompt_char_{$__fish_svn_prompt_flag_names[$flag_index]}_color
+			# set the colour and print display character, then restore to default display colour
+			printf '%s%s%s' (set_color $$flag_color) $$flag_display (set_color normal)
+		end
+	end
+end
+
+
+function __fish_svn_prompt --description "Prompt function for svn"
+	# if svn isn't installed then don't do anything
+	if not command -s svn > /dev/null
+		return 1
+	end
+
+	# make sure that this is a svn repo
+	set -l checkout_info (command svn info ^/dev/null)
+	if [ $status -ne 0 ];
+		return
+	end
+
+	# get the current revision number
+	set -l repo_revision_number (command svn info | grep "Last Changed Rev: " | sed "s=Last Changed Rev: ==")
+	printf '(%s%s%s' (set_color $__fish_svn_prompt_color_revision) $repo_revision_number (set_color normal)
+
+	# resolve the status of the checkout
+	# 1. perform `svn status`
+	# 2. remove extra lines that aren't necessary
+	# 3. cut the output down to the first 7 columns, as these contain the information needed
+	# 4. replaces newline characters with ":" to work around multi-line string handling
+	set -l svn_status_lines (command svn status | sed -e 's=^Summary of conflicts.*==' -e 's=^  Text conflicts.*==' -e 's=^  Property conflicts.*==' -e 's=^  Tree conflicts.*==' -e 's=.*incoming .* upon update.*==' | cut -c 1-7 | tr '\n' ':')
+
+	# track the last column to contain a status flag
+	set -l last_column 0
+
+	# iterate over the 7 columns of output (the 7 columns are defined on `svn help status`)
+	for col in (seq 7)
+		# get the output for a particular column
+		# 1. echo the whole status flag text
+		# 2. swap the ":" back to newline characters so we can perform a column based `cut`
+		# 3. cut out the current column of characters
+		# 4. remove spaces and newline characters
+		set -l column_status (echo $svn_status_lines | tr ':' '\n' | cut -c $col | tr -d ' \n')
```

Comment:
Interesting. OK i will look at changing that. I remember reading there was some funky-ness with handling multi-line strings in a variable in fish shell (and experiencing some of that myself). And came to this code as it "worked". Thanks!